autoload/maktaba/plugin.vim

```diff
@@ -303,14 +336,7 @@ endfunction
 " then maktaba can't handle a plugin named "plugins/myplugin". Make sure your
 " plugins have sufficiently different names!
 function! maktaba#plugin#CanonicalName(plugin) abort
-  let l:plugin = a:plugin
-  if maktaba#string#StartsWith(l:plugin, 'vim-')
-    let l:plugin = l:plugin[4:]
-  endif
-  if maktaba#string#EndsWith(l:plugin, '.vim')
-    let l:plugin = l:plugin[:-5]
-  endif
-  return l:plugin
+  return matchstr(a:plugin, '\m\C^\(vim-\)\?\zs.\{-}\ze\(\.vim\)\?$')
```

Comment:
Could also use `'\v\C^(vim-)?\zs.{-}\ze(\.vim)?$'`, which might be a bit more readable (and presumably as performant).

Perhaps add a note somewhere that this function is performance-sensitive?